src/app/AlwaysOn.HealthService/HealthJob.cs

```diff
@@ -0,0 +1,66 @@
+﻿using AlwaysOn.Shared;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AlwaysOn.HealthService
+{
+    /// <summary>
+    /// Background job that performs the health checks on a regular interval and caches the result in memory
+    /// Initial source: https://stackoverflow.com/a/68630985/1537195
+    /// </summary>
+    public class HealthJob : BackgroundService
```

Comment:
Just for me to understand: There was already caching implemented in the previous version. What's the motivation to move to a background job, when the result is used solely by the controller?

Reply:
a) It does consistent checking, not dependent on any external calling it every x seconds
b) It doesnt require any actual Cache, it just uses static objects which makes it much easier to understand

Reply:
I'd say it is a cleaner design this way. Plus, the backends get checked consistently not just when somebody decides to call the health service. This way every call to the health service should be equally fast. Not some calls taking longer because a check is now being executed( because cache might have expired). Also I think on the previous design simulatanious calls could have realized that a cache value expired and trigger new downstream checks in parallel

Reply:
After a call I agree.